src/main/kotlin/app/configuration/JobConfiguration.kt

```diff
@@ -20,6 +19,7 @@ import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.context.annotation.Profile
 import org.springframework.core.task.SimpleAsyncTaskExecutor
+import org.springframework.retry.policy.SimpleRetryPolicy
```

Comment:
we don't appear to be using this in our other code, is there a reason?

Reply:
In our other code we put the retry annotation (akin to python decorators) on individual functions - which is what I tried on the first attempt. However what I think is happening here is that the retry tried to retry on an input  stream that had already been read from. So this attempt just restarts the whole processing of the file from the start, i.e. the whole spring batch step is retried - which should reinitialise the input stream.

Reply:
apologies for merging before answering.